sonic_platform_base/sonic_xcvr/api/public/c_cmis.py

```diff
@@ -257,8 +260,26 @@ def get_pm_all(self):
         PM_dict['rx_mer_avg'] = self.xcvr_eeprom.read(consts.RX_AVG_MER_PM)
         PM_dict['rx_mer_min'] = self.xcvr_eeprom.read(consts.RX_MIN_MER_PM)
         PM_dict['rx_mer_max'] = self.xcvr_eeprom.read(consts.RX_MAX_MER_PM)
+
+        self.xcvr_eeprom.write(consts.VDM_CONTROL, VDM_UNFREEZE)
+        #Retry for freeze done status
+        if self.util_vdm_freeze_unfreeze_status_retry(VDM_FREEZE_UNFREEZE_DONE_RETRY, SLEEP_50MS, consts.VDM_UNFREEZE_DONE, 1):
+            helper_logger.log_error('UnFreeze done status check timeout expired')
+
         return PM_dict
 
+    def util_vdm_freeze_unfreeze_status_retry(self, retry, sleep_time, reg_field, expected_status):
```

Comment:
could we add a short description for this method as well as the explanation for return code?

Reply:
added